docs/src/main/java/org/geotools/tutorial/coverage/ImageTiler.java

```diff
@@ -0,0 +1,224 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+//docs start prelim
+package org.geotools.tutorial;
+
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.coverage.grid.io.GridFormatFinder;
+import org.geotools.coverage.processing.CoverageProcessor;
+import org.geotools.coverage.processing.Operations;
+import org.geotools.factory.Hints;
+import org.geotools.gce.geotiff.GeoTiffFormat;
+import org.geotools.geometry.Envelope2D;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.resources.Arguments;
+import org.opengis.geometry.Envelope;
+import org.opengis.parameter.ParameterValueGroup;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Simple tiling of a coverage based simply on the number vertical/horizontal tiles desired and
+ * subdividing the geographic envelope. Uses coverage processing operations.
+ *
+ */
+public class ImageTiler {
+
+    private final int NUM_HORIZONTAL_TILES = 16;
+    private final int NUM_VERTICAL_TILES = 8;
+
+    private Integer numberOfHorizontalTiles = NUM_HORIZONTAL_TILES;
+    private Integer numberOfVerticalTiles = NUM_VERTICAL_TILES;
+    private Double tileScale;
+    private File inputFile;
+    private File outputDirectory;
+
+
+    private String getFileExtension(File file) {
```

Comment:
Is there a DataUtilities method that does this (very common) operation? I would like to avoid the use of boilerplate java code in our tutorials - anything of that nature gets made into a DataUtilities method (or similar).